### ISSUE TITLE
Incorrect 404 NotFound when two route handlers share the same prefix

### ISSUE THREAD

**Issue by ottermata** [starlite-api/starlite#829]
**Describe the bug**
When a route handler extends another with path parameters, the shorter one cannot be called.

**To Reproduce**
`GET /foo` will result in a 404:
```
from starlite import Starlite, get


@get('/{foo:str}')
def print_foo(foo: str) -> None:
    print(foo)


@get('/{foo:str}/{bar:str}')
def print_foo_bar(foo: str, bar: str) -> None:
    print(foo, bar)


app = Starlite(
    [
        print_foo,
        print_foo_bar
    ],
)
```

**Comment by peterschutt** [comment]
Can you check this on main, I think maybe this is same as #816 and (hopefully) fixed in #818.

**Comment by ottermata** [comment]
You are right, my bad!
